adapter/echo/middleware.go

```diff
@@ -0,0 +1,41 @@
+package echo
+
+import (
+	"net/http"
+
+	sentinel "github.com/alibaba/sentinel-golang/api"
+	"github.com/alibaba/sentinel-golang/core/base"
+	"github.com/labstack/echo/v4"
+)
+
+// SentinelMiddleware
+func SentinelMiddleware(opts ...Option) echo.MiddlewareFunc {
+
+	options := evaluateOptions(opts)
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) (err error) {
+			resourceName := c.Request().Method + ":" + c.Request().URL.Path
```

Comment:
Can we extract the matching API pattern (e.g. `/foo/:id`) instead of the actual URL?